api/types.ts

```diff
@@ -7,12 +7,12 @@ export interface BookInterface {
   synopsis: string
 }
 
-export interface UserInterface {
+export type UserType = {
   id: number
   username: string
   auth: string
 }
 
-export interface BooksInterface extends Array<BookInterface> {
-  books: BookInterface[]
+export interface Books extends Array<BookType> {
```

Comment:
I think you can just use as using this will just make you end up using `extends` again. Can you try it out and see if this works?:

```
interface ComponentProps {
  books: Array<BookType> 
}
```